traits/has_traits.py

```diff
@@ -396,6 +422,13 @@ def update_traits_class_dict(class_name, bases, class_dict):
     # Move all trait definitions from the class dictionary to the
     # appropriate trait class dictionaries:
     for name, value in list(class_dict.items()):
+
+        # Warn name collisions.
+        if check_name_collision and _is_disallowed_prefix(name):
+            warnings.warn("The attribute named '{}' of class {} uses a "
+                          "prefix that is discouraged. Consider renaming it."
+                          .format(name, class_name), TraitsWarning, 3)
```

Comment:
The users will likely want to know exactly what prefixes should not be used are so they don't have to fix this by trial-and-error. Please mention the protected prefixes in the warning message.